hw3_2/hw3-2/Program.cs

```diff
@@ -0,0 +1,452 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace hw3_2
+{
+    class Program
+    {
+        static async Task Main(string[] args)
+        {
+            string path;
+            string argument;
+            if (args.Length != 2)
+            {
+                Console.WriteLine(
+                    "Few arguments! Enter file path and \"-c\" if you want to compress text\nor \"-u\" if decompress");
+                return;
+            }
+
+            path = args[0];
+            argument = args[1];
+            if (argument != "-c" && argument != "-u")
+            {
+                Console.WriteLine(
+                    "Incorrect argument. Enter file path and \"-c\" if you want to compress text\nor \"-u\" if decompress");
+                return;
+            }
+
+            if (!File.Exists(path))
+            {
+                Console.WriteLine("Incorrect file path!");
+                return;
+            }
+
+            Lzw lzw = new();
+            if (argument == "-c")
+            {
+                string? text = "";
+
+                using (StreamReader reader = new StreamReader(path))
+                {
+                    if ((text = await reader.ReadToEndAsync()) == null)
+                    {
+                        Console.WriteLine("Empty file!");
+                        return;
+                    }
+                }
+
+                bool isEmpty;
+                List<Byte[]> listOfCodes = lzw.LzwCompression(text, out isEmpty);

```

Comment:
```suggestion
                (List<Byte[]> listOfCodes, bool isEmpty) = lzw.LzwCompression(text);
```
Не используйте `out` параметры. Вместо этого просто верните пару